src/corehost/cli/deps_resolver.cpp

```diff
@@ -397,11 +397,6 @@ bool deps_resolver_t::resolve_tpa_list(
     };
 
     // First add managed assembly to the TPA.
-    // TODO: Remove: the deps should contain the managed DLL.
-    // Workaround for: csc.deps.json doesn't have the csc.dll
```

Comment:
Looking at the latest SDK, it looks like csc.deps.json still doesn't have the csc.dll/.exe.

Are we sure we want to break this?

Reply:
No I was just asserting until I found a reason not to. I will keep this in. Thanks

Reply:
FYI, it looks like in the absolute latest SDK that the Roslyn pieces have changed, and now csc.deps.json does have a csc.dll listing.  But I think this new host will still need to be able to launch already released SDKs (since users can pin an SDK version with a global.json file).

Reply:
I've updated the code to include explicit checks for duplicates, and left the original work-around in.